### DISCUSSION TITLE
Sort order of comments on the Conversation tab

### DISCUSSION THREAD

**Discussion by carsten-krueger** [community/community#129980] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

In a pull request on the Conversation tab it would be good to be able to change the sort order of comments.
Currently comments are sorted in ascending order from oldest to newest. I'd like to sort comments in descending order so that the newest comment is on top. As far as I know there is currently no way to do that.

**Comment by vishveshjain** [comment]
You're right, unfortunately there is no built-in way to sort pull request comments directly on the GitHub Conversation tab. By default, they are displayed in ascending order, with the oldest comment at the top.

Here are a couple of workarounds to achieve your desired view:

1. **Manual Scrolling:** This might not be ideal for long comment threads, but you can simply scroll down to the bottom of the conversation to see the newest comments.

2. **Browser Extension:** There might be browser extensions that offer comment sorting functionalities for various platforms.  A quick search for "sort github comments" or "sort conversation tab comments" might reveal some options depending on your browser.

While there's no native sorting option within GitHub, hopefully these methods can help you see the newest comments in your pull request conversations.

**Comment by roshan-rubrik** [comment]
This is very much needed.

**Comment by RobertTheBlair** [comment]
When we get this feature, please have it be a sticky preference on user settings.   (along with other things, like default pref for whitespace in diffs)

**Comment by tfactor2** [comment]
Here is a simple TamperMonkey script to get the job done:
```js
// ==UserScript==
// @name         github-pr-conversation-reversed
// @version      0.1
// @description  Reverses the order of conversation (from the recent to the oldest one)
// @author       tfactor2
// @match        https://github.com/*/pull/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=github.com
// @grant        none
// ==/UserScript==

(function reverseConversations() {
    const $discussionDiv = document.querySelector('.js-discussion');
    if ($discussionDiv && !$discussionDiv.hasAttribute('data-github-pr-conversation-reversed-processed')) {
        const children = Array.from($discussionDiv.children);
        children.reverse().forEach(child => $discussionDiv.appendChild(child));
        $discussionDiv.setAttribute('data-github-pr-conversation-reversed-processed', 'true');
    }
    setTimeout(reverseConversations, 1000);
})();
```

**Comment by AaronMachin** [comment]
+1 - why do I need to scroll to the bottom of the page to see the most recent information?

**Comment by fixedforward** [comment]
Gitlab has this option so I think Github is behind here ....

**Comment by ltpitt** [comment]
We really need this... Copilot can fix it in a sec.

**Comment by tickytacky** [comment]
We also used this feature in gitlab (it's a user setting) and after migration to github/ghe each time run crazy when we have to scroll to the bottom in our heavily commented Issues. Would be interesting to track how much time we waste with scrolling down, in our large projects...
Basically I think it's more important to have the newest comments on top, simply changing the sort order to descending would be fine for us ;-)
And this applies not only for PRs but (for us more important) for Issues too!

**Comment by saikatbangla** [comment]
GitHub does not at the moment have a native option for reversing the sort order of the comments on the Pull Request Conversation tab. The comments and the timeline events remain in chronological order (from oldest to newest).
> ---
> 
> 
> **Recommended Workarounds**
> * **Tampermonkey / Violentmonkey Userscript**
> Reverse the conversation timeline DOM elements automatically on page load:
> ```javascript
> // ==UserScript==
> // @name         GitHub PR Newest Comments First
> // @match        https://github.com/*/*/pull/*
> // @grant        none
> // ==/UserScript==
> 
> (function reversePR() {
const container = document.querySelector('.js-discussion');
if (container && !container.dataset.reversed) {
The items are obtained by calling Array.from on the children of the container.
>         items.reverse().forEach(item => container.appendChild(item));
>         container.dataset.reversed = 'true';
>     }
>     setTimeout(reversePR, 1000);
> })();
> 
> ```
> 
> 
> * **Browser Extensions**
**Refined GitHub:** Includes features that allow for quick navigation through the timeline, filtering of activity, and direct access to the most recent comments.
* **PR Latest First:** The order of the PR conversation streams is set so that the most recent activity appears first.
> 
> 
> * **Keyboard Shortcuts & Timeline Filters**
Press the 'End' key (or use 'Cmd + Down Arrow' on a Mac) to move directly to the most recent comment at the bottom.
You can use the dropdown menu for the Timeline filter at the top of the PR discussion to hide the bot's activity, the commits, and the status checks, which in turn greatly reduces the amount you need to scroll.
> 
> 
> 
>
